shared/chroot.go

```diff
@@ -222,7 +224,19 @@ func SetupChroot(rootfs string) (func() error, error) {
 				Value: "noninteractive",
 				Set:   true,
 			},
-		})
+		}
+
+	} else {
```

Comment:
The idea is maintains compatibility with current execution and where it's needed define list all envs.

Do you means that If i want override default values i could set for example TERM and override default value?

Reply:
The problem here is that if I want override an existing value I need iterate between all array entries instead of simplify my work on define a list of environment variables.
In additional to this why set always DEBIAN_FRONTEND variable ? Not needed on centos, gentoo, sabayon and others distro.
I don't like iterate an array list for override a variable and i think that also if i add two time the same variable it's bad solution.
wdyt ?

Reply:
> The problem here is that if I want override an existing value I need iterate between all array entries instead of simplify my work on define a list of environment variables.

I was thinking, we could use `map[string]EnvVariable` for the env variables code. This would make more sense (keys can only be set once anyway), and simplify your code. Here's the idea:
```go
type EnvVariable struct {
	Value string
	Set   bool
}

type Environment map[string]EnvVariable

func SetupChroot(rootfs string, envs []DefinitionEnvVars) (func() error, error) {
	...
	env := Environment{
		"PATH": EnvVariable{
			Value: "/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin",
			Set:   true,
		},
		"SHELL": EnvVariable{
			Value: "/bin/sh",
			Set:   true,
		},
		"TERM": EnvVariable{
			Value: "xterm",
			Set:   true,
		},
		"DEBIAN_FRONTEND": EnvVariable{
			Value: "noninteractive",
			Set:   true,
		},
	}

	// This would be your code
	for _, e := range envs {
		_, ok := env[e.Key]
		if ok {
			env[e.Key].Value = e.Value
		} else {
			env[e.Key] = EnvVariable{
				Value: e.Value,
				Set:   true,
			}
		}
	}

	// Set environment variables
	oldEnv := SetEnvVariables(env)
	...
}
```

> In additional to this why set always DEBIAN_FRONTEND variable ? Not needed on centos, gentoo, sabayon and others distro.

If it's not used, it doesn't really matter whether it's set or not.

> I don't like iterate an array list for override a variable and i think that also if i add two time the same variable it's bad solution.

With the `map` solution, variables/keys cannot be added multiple times.

What do you think?

// cc: @stgraber

Reply:
> If it's not used, it doesn't really matter whether it's set or not.
It seems not a good response.

About `map` sound good.

Reply:
The map has been implemented. Regarding the default variables, we could add a new key `clear_defaults` (or something like that) to the `environment` structure. This way, everyone should be happy.
```go
type DefinitionEnvVars struct {
	Key   string `yaml:"key"`
	Value string `yaml:"value"`
}

type DefinitionEnv struct {
	ClearDefaults bool                `yaml:"clear_defaults,omitempty"`
	EnvVariables  []DefinitionEnvVars `yaml:"variables,omitempty"`
}

type Definition struct {
	Environment DefinitionEnv `yaml:"environment,omitempty"`
}
```

Reply:
Done. Tell me if I lost something